In [5]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [25]:
df = pd.read_excel('dataset_soal2.xls')
df.head()

Category  Weather V-1  Holiday V-2  Game V-3  Qty
0        A            5            1         0  250
1        B            3            1         1  200
2        C            1            1         0   75
3        D            4            1         1  400
4        E            4            0         0  150

# 2A

In [26]:
import math 
dis = []
for i in range(6):
    dis.append(math.sqrt((float(df.iloc[i]['Weather V-1'])-1)**2
                         +(float(df.iloc[i]['Holiday V-2'])- 0)**2
                         +(float(df.iloc[i]['Game V-3'])-0)**2))

df['dis'] = dis
df

Category  Weather V-1  Holiday V-2  Game V-3  Qty       dis
0        A            5            1         0  250  4.123106
1        B            3            1         1  200  2.449490
2        C            1            1         0   75  1.000000
3        D            4            1         1  400  3.316625
4        E            4            0         0  150  3.000000
5        F            2            0         0   50  1.000000

In [13]:
df.sort_values('dis')

Category  Weather V-1  Holiday V-2  Game V-3  Qty       dis
2        C            1            1         0   75  1.000000
5        F            2            0         0   50  1.000000
1        B            3            1         1  200  2.449490
4        E            4            0         0  150  3.000000
3        D            4            1         1  400  3.316625
0        A            5            1         0  250  4.123106

In [32]:
y = df.sort_values('dis').head(1)

In [33]:
x = y["Qty"]
x

2    75
Name: Qty, dtype: int64

In [34]:
np.mean(x)

75.0

Apabila Cuaca buruk dengan nilai = 1, Weekday, dan Game = 0, maka roti yang harus dibuat sebanyak 75 buah

In [35]:
df.to_excel('D:/Campus/Outputs2Abaru.xls')

# 2B

In [23]:
import math 
dis = []
for i in range(6):
    dis.append(math.sqrt((float(df.iloc[i]['Weather V-1'])-4)**2
                         +(float(df.iloc[i]['Holiday V-2'])- 1)**2
                         +(float(df.iloc[i]['Game V-3'])-1)**2))
df['dis'] = dis
df

Category  Weather V-1  Holiday V-2  Game V-3  Qty       dis
0        A            5            1         0  250  1.414214
1        B            3            1         1  200  1.000000
2        C            1            1         0   75  3.162278
3        D            4            1         1  400  0.000000
4        E            4            0         0  150  1.414214
5        F            2            0         0   50  2.449490

In [24]:
df.sort_values('dis').head(4)

Category  Weather V-1  Holiday V-2  Game V-3  Qty       dis
3        D            4            1         1  400  0.000000
1        B            3            1         1  200  1.000000
0        A            5            1         0  250  1.414214
4        E            4            0         0  150  1.414214

In [28]:
x = df.sort_values('dis').head(4)
y = x["Qty"]
y

2     75
5     50
1    200
4    150
Name: Qty, dtype: int64

In [30]:
np.mean(y)

118.75

Apabila Cuaca baik dengan nilai 4, Weekend, dan Game =1, maka roti yang harus dibuat sebanyak 119 buah

In [36]:
df.to_excel('D:/Campus/Outputs2Bbaru.xls')